### ISSUE TITLE
Allow display of gameless systems

### ISSUE THREAD

**Issue by digitalLumberjack** [digitalLumberjack/recalbox-os#294]
Add a switch in ES that display or not all systems, even if no game is detected

**Comment by nickdj** [comment]
Will the Raspberry Pi 2 have enough memory to do this?  When more systems are displayed it uses more VRAM.  Somewhere around 160MB I get a white screen when using a high-res theme.
Maybe a warning message should be displayed before enabling if the user not using the default theme?

**Comment by customcarvin** [comment]
I agree with nickdj, ES is already enough of memory hog! Actually, it would be nice to enable/disable backgrounds for each system or switch with a "generic" recalbox background to be used for every system in order to conserve memory; especially for those of us on 256mb pi ;-)

This might be more of an ES issue and maybe not something retroboy wants to bother with (wouldn't blame him). I myself have noticed that if I enable any more than 7 systems at 720p (mode=4 - 128/128) ES runs out of ram and the "8th system" will not display images. Then once I open any system and begin to browse through games, ES eventually corrupts and crashes. All of my covers are small compressed jpgs and pngs no larger than 30kib, but it all adds up quickly :-)

**Comment by nickdj** [comment]
I created a new theme which had both 1080p and 720p resolutions and managed to get the latter to use less than the default theme!  However, since then I've created another with mostly a flat colours and shading and that uses even less.  The image size of the files makes very little difference, it's actually the resolution and amount of colours being displayed which consumes memory.  It is very possible for me to make a super efficient theme which would run all systems and still use less than 120mb.  I suspect if I remove background music it will further reduce the overhead.  Will see what I can do.  😃

**Comment by customcarvin** [comment]
Sounds good nickdj! I'm excited to see what you come up with.

**Comment by nickdj** [comment]
https://www.mediafire.com/folder/ypp92a26q516k/Recalbox

'clean-720' will run 20 systems and only consume 107Mb of VRAM.  Music files make no difference to memory usage and compressing images has little impact.  Will play with reducing each image to 4 colours/shades (no smooth shading) to see what effect that has on VRAM. :)

**Comment by customcarvin** [comment]
Very nice work nickdj! Thank you very much! I'll let you know how the clean-720 goes as soon as I have a chance. :smiley:
